examples/nextjs/pages/_document.js

```diff
@@ -58,22 +52,26 @@ MyDocument.getInitialProps = async (ctx) => {
   // 4. page.render
 
   const originalRenderPage = ctx.renderPage;
+  // You can consider sharing the same emotion cache between all the SSR requests to speed up performance.
+  // However, be aware that it can have global side effects.
+  const cache = createEmotionCache();
 
-  const cache = getCache();
   const { extractCriticalToChunks } = createEmotionServer(cache);
 
   ctx.renderPage = () =>
     originalRenderPage({
       // Take precedence over the CacheProvider in our custom _app.js
-      enhanceComponent: (Component) => (props) =>
+      enhanceApp: (App) => (props) =>
         (
           <CacheProvider value={cache}>
-            <Component {...props} />
+            <App disableEmotionCache {...props} />
```

Comment:
Could we instead pass the `cache` to `App` and then `App` uses the passed `cache` or otherwise use the client-side cache?

```js
// _document.js
<App cache={cache} {...props} />

// _app.js
// Client-side cache, shared for the whole session of the user in the browser.
const clientSideEmotionCache = createEmotionCache();
export default function MyApp(props) {
  const { Component, pageProps } = props;
  const { Component, emotionCache = clientSideEmotionCache, pageProps } = props;

  return <CacheProvider value={emotionCache}>...</CacheProvider>
}
```

That way we have the same component tree on client and server and just the origin of the `cache` changes. 

Not blocking for this PR but if this is possible it'd be nice to get this simplification in.

Reply:
Great point. I don't see why it couldn't work and it would be simpler.

Reply:
I have tried implementing this approach